src/collection/collection.service.ts

```diff
@@ -241,6 +244,65 @@ export class CollectionService {
 		return get(collection, 'collection_fragments') || [];
 	}
 
+	/**
+	 * Find name that isn't a duplicate of an existing name of a collection of this user
+	 * eg if these collections exist:
+	 * copy 1: test
+	 * copy 2: test
+	 * copy 4: test
+	 *
+	 * Then the algorithm will propose: copy 3: test
+	 * @param prefix
+	 * @param existingTitle
+	 * @param user
+	 */
+	public static getCopyTitleForCollection = async (
+		prefix: string,
+		existingTitle: string,
+		user: Avo.User.User
+	): Promise<string> => {
+		const collections = await CollectionService.getCollectionTitlesByUser('collection', user);
+		const titles = collections.map(c => c.title);
+
+		let index = 0;
+		let candidateTitle: string;
+		do {
+			index += 1;
+			candidateTitle = prefix.replace('%index%', String(index)) + existingTitle;
+		} while (titles.includes(candidateTitle));
+
+		return candidateTitle;
+	};
+
+	public static async duplicateCollection(
+		collection: Avo.Collection.Collection,
+		user: Avo.User.User,
+		prefix: string,
+		triggerCollectionInsert: any,
+		triggerCollectionFragmentsInsert: any
+	): Promise<Avo.Collection.Collection> {
+		collection.owner_profile_id = getProfileId(user);
+		collection.is_public = false;
+		delete collection.id;
```

Comment:
Die collection die hier wordt doorgegeven is een state value dus best oppassen met gwn properties te deleten om bugs te vermijden.

Reply:
goed punt
cloned before changes: https://github.com/viaacode/avo2-client/commit/cc4a0e700cd43b54c93005748dd5490713815ef0